main/autocode/src/org/ejml/GenerateJavaCode32.java

```diff
@@ -54,6 +54,14 @@ public GenerateJavaCode32() {
         prefix32.add("FUnary");
         prefix64.add("DBinary");
         prefix32.add("FBinary");
+        prefix64.add("DMonoid");
+        prefix32.add("FMonoid");
+        prefix64.add("DSemiRing");
+        prefix32.add("FSemiRing");
+        prefix64.add("DMonoids");
+        prefix32.add("FMonoids");
+        prefix64.add("DSemiRings");
+        prefix32.add("FSemiRings");
```

Comment:
Just curious if you are planning to add `int` and `boolean` domains to this PR?

Reply:
No not in this PR, I think this should belong in a seperate PR.
However currently I don't plan to add more types of matrices, except for sparse boolean masks in another PR (implictly true for every defined cell).

I see the main problem with different types in having to support variants of all operators to support all combinations of types.
Guess this could also be handled by code generation, but requiring significant amout of work in my opinion.

Reply:
There might be a way to avoid unnecessary duplication by using generics, but it would require a different API and would probably be more difficult to optimize on the JVM. Agree it makes sense to have a separate PR for #68.

Reply:
The problem I know of, is that generics don't support primitive types.
I think project valhalla solves this problem, so maybe in a future JDK it will be possible. 
Also the return values from f.i. get() couldn't be primitives then.